### DISCUSSION TITLE
Playtime 2 is changing my kick drum transient sound at the start of each loop. (The same source audio works perfectly in Playtime 1.)

### DISCUSSION THREAD

**Discussion by FakenameMusic** [helgoboss/helgobox#1142] [Q&A] [answered]
At the outset it might help to know that I have tried to reproduce this Playtime 2 issue in Playtime 1, with the exact same source audio file, in the same Reaper session - by loading an instance of both Playtime 1 and Playtime 2 in the same Reaper session, but I cannot, as Playtime 1 handles it perfectly and as expected: 4 identical kick drums.

So...
I have made a 1 bar kick drum loop - a simple "4-to-the-floor" pattern by using the "Import selected items into slot" function. The pattern is made from 4 identical Kick Drum samples. Unfortunately at the start of each bar/loop the kick drum sounds different when imported into Playtime. The transient of the first kick has been altered. The first kick in each bar sounds softer, I believe because Playtime has removed/faded the first few milliseconds of the transient - so it sounds considerably less "clicky" than the other kicks in the bar, where the desirable "click" of the kick has not been modified.

I checked that the source audio has no auto "fade in" in Reaper (I don't know if Playtime cares about this) but there is no fade in as can be seen here:

![NoFade in original audio](https://github.com/user-attachments/assets/20022c06-a35e-46f7-a55d-725bbb0fe650)

When I double click for the waveform view in Playtime 2 (I know this is still very much a beta feature) I can see that the first part of the waveform *looks* different - almost like it has been cut/faded as I suspected- see here:

![2 - Playtime Wav Display](https://github.com/user-attachments/assets/bd16e763-08fc-4d7a-9a1e-7d89d07b2d6b)

As a test I tried importing a WAV file directly by drag and drop, but unfortunately the issue is the same.

Here's an image showing how, visibly, the transient of the first kick has been changed and is smoother than the others in the bar. The other 3 kicks still have the wobbly line and sudden rise that represents the "click" of the kick, and they sound exactly as they should be, and as I want them:

![bad kick](https://github.com/user-attachments/assets/84ee951a-a642-44ac-870f-4899cfd23bdc)

Here is the audio file as it should sound:
https://sndup.net/tcxjv/d

Here is Playtime's version where audibly every 4th kick drum (at the start of a bar/loop) sounds different.
https://sndup.net/gmnqk/d

This is a really big problem, and makes Playtime 2 unusable for me. :-/

**Comment by helgoboss** [comment]
Damn, I should really get on writing the docs. 

You need to switch off the "Source fades" checkbox: <img width="281" alt="image" src="https://github.com/user-attachments/assets/a72cbbdb-5669-44e0-a1ec-cf801f287226">

Playtime automatically enables "Source fades" because it's desired in most cases, otherwise everyone would complain about clicks. Maybe I should not set is as default when doing imports from files. And when importing from an item, it would ideally inherit the item fade length. If you agree, I will do this ASAP, it's not a biggie.

**Reply by FakenameMusic** [reply]
Hi, the "sources fade" does not work for me I'm afraid, I still have the issue: I had already tried "sources fade" as I had found it while scanning through options in the side panel for each individual cell. But it didn't do anything, I though maybe I was misunderstanding this feature, so I didn't mention it.

Regarding documentation: I think you have done a good job with the in-UI notes when you hold the mouse over a control: a curious exploring user can find the info you have put in - just as you were saying in a post elsewhere. Sorry I have been a pain in your ass today!

But yes I agree, your suggested way of implementing it is good. No fade for "external" samples. And inheriting the fade length from the media item would make sense, as even a basic Reaper user would know how to change the fade length. It would offer great flexibility, and you wouldn't have to implement user-adjustable fades in your UI.

You could however still keep a "default length" over-ride on the button too to appease a less aware user?
Perhaps even make it a 3 way toggle like your "light/dark/auto" theme button - Fade OFF/DEFAULT/INHERIT to keep the user in control?It's such important parameter in every individual cell though, that perhaps the option should be made more visible - perhaps even being in the cell's right-click menu?

But for starters, even just the simple thing you suggested would be great.

Unfortunately though it hasn't solved my issue. Unless of course the existing toggle button is currently broken for some reason?

**Reply by helgoboss** [reply]
Please send me the project directory in question (must be self-contained of course) along with a few instructions which clip to trigger. I will look into it.

**Reply by dantec3p** [reply]
Hi Helgo, is there any way I could help you on writing the docs? Please ask if you need. Would be glad...

**Reply by helgoboss** [reply]
Hi @dantec3p! Thanks for the offer. You mean Playtime docs in general? The technical docs is probably something that I need to do myself. But if you want to add tutorial-style docs (wiki), that would be awesome.

**Comment by FakenameMusic** [comment]
Here is the Reaper session. I made it as simple / small as possible.
If you launch Playtime and play the loop you will hear the discrepancy. There is only 1 active cell: Column 1, Row 1.

Track 1 "Kick Audio" is the source file for the Playtime loop. If you unmute the track, hit "1" to go to the marker point, and then Play, you will hear the 4 kicks play in a loop, with a little time before the first kick so that you can hear there is no issue with the 1st transient in the source file.

[Helgoboss Error.zip](https://github.com/user-attachments/files/16786709/Helgoboss.Error.zip)

**Reply by helgoboss** [reply, marked as answer]
Thanks.

Okay, it's what I thought. The item doesn't start with the beginning of the audio file. It starts half a beat after (0.5 beats). Now, when you import this into Playtime, it will respect that item cut and do the same: It sets the static **section** start position to 0.5 beats as well. So far, so good.

Now the issue is that Playtime realizes that 0.5 beats is a position somewhere **within** the material and assumes that it's safer to introduce a fade, in order to prevent audible clicks due to sudden amplitude changes. This is called a **section fade** in Playtime because it happens at the beginning of a section (not at the beginning of the complete source, which would be the "source fade" that we have talked bout). Section fades can't be disabled currently. However, I see now that there are cases where it makes sense to disable section fades as well. On import from item, it should ideally apply the fade length of the item, in your case no fades at all. Which leaves "Interaction fades" as the only kind of fades that should **always** be applied (whenever playback starts or ends **within** a section).

I will think about this. You just made the next Helgobox release take a bit longer hehe ;)

BTW, workaround in your case is to glue the WAV file in REAPER with its current cut, import it into Playtime again and switch off "Source fades".

**Reply by FakenameMusic** [reply]
OK great, that work around works - so glad to have a work around. Thanks for all your work and help. 
 Should I close this discussion or leave it open as you've said it still needs work? Whatever is most helpful to you?

**Reply by helgoboss** [reply]
Great!

Q&A discussions are supposed to be left open, so that other people can look up the answers as well. It would be cool if you could mark something as an answer though. Thanks!

I think I came up with a good way to deal with this in a better way. At least it makes sense to me at the moment ;)

See #1146

No need to comment on that, unless you see something weird (this is more like a todo list for myself).

**Reply by helgoboss** [reply]
Will do that in https://github.com/helgoboss/helgobox/issues/1146

**Comment by FakenameMusic** [comment]
Great thank you very  much for your help.

Perhaps there should be a toggle in the setting panel in the Playtime UI to set the desired default behaviour for the fade? Depending on which way you work you may *always* want a specific type of fade behaviour?

Also perhaps there should be some disambiguation of "clip" and "source"? Am I incorrect that saying these are actually the same thing?

Perhaps a wording change on the controls themselves would make the function clearer too?

![static source](https://github.com/user-attachments/assets/723ec7e4-b661-4eac-bed9-73e7c122bb38)

"Delay source start" because there will never be a negative value. 
"Pad ending" because you're telling the user that something will be added at the end?

Just some suggestions.
Thanks!

**Reply by helgoboss** [reply]
> Great thank you very much for your help.
> 
> Perhaps there should be a toggle in the setting panel in the Playtime UI to set the desired default behaviour for the fade? Depending on which way you work you may _always_ want a specific type of fade behaviour?

What types of fade are you referring to?

> Also perhaps there should be some disambiguation of "clip" and "source"? Am I incorrect that saying these are actually the same thing?

"Source" is the underlying audio file. "Clip" is the usage of that audio file. It's a similar terminology in REAPER: "Source" and "Item".

> Perhaps a wording change on the controls themselves would make the function clearer too?
> 
> ![static source](https://private-user-images.githubusercontent.com/178907884/363158708-723ec7e4-b661-4eac-bed9-73e7c122bb38.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMjQ1OTMsIm5iZiI6MTcyNTAyNDI5MywicGF0aCI6Ii8xNzg5MDc4ODQvMzYzMTU4NzA4LTcyM2VjN2U0LWI2NjEtNGVhYy1iZWQ5LTczZTdjMTIyYmIzOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMFQxMzI0NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03NTk1OTNiMjQ2NjNkYTk1YzFkYzE2NDM3OTAyZDJjOWZmMjM0ZmY0ZjkzMmYxYTIwZDA4NjJhMTU2OTk3YzI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ODW9VhNbW7bYuMJLXbebJo8IKhRpn8rcfNNcKXJggrg)
> 
> "Delay source start" because there will never be a negative value. "Pad ending" because you're telling the user that something will be added at the end?

That would be an unusual terminology and misleading. Nothing is delayed here.

**Reply by FakenameMusic** [reply]
OK, all good. It makes sense to me, but you are the one with the full knowledge and overview in the project.

And he Source/Clip Source/Item makes sense to me now thanks.

The fades I was referring to were the source fades. For users like me who prefer to manage their own fades, or others who will only work with premade loops, it would be good to have the option of "always off" for the source fade.

**Reply by helgoboss** [reply]
Now I understand. Could be a possible addition for the future. I won't add that right away. If you need it, please create a feature request (= discussion in the "Ideas" category).
